.github/workflows/mirror-images-regsync.yml

```diff
@@ -0,0 +1,64 @@
+            echo "manual_secrets=true" >> $GITHUB_OUTPUT
+          elif [ -n "${{ secrets.DEBUG_APPCO_USERNAME }}" ]; then
+            echo "manual_secrets=true" >> $GITHUB_OUTPUT
+          elif [ -n "${{ secrets.DEBUG_APPCO_PASSWORD }}" ]; then
+            echo "manual_secrets=true" >> $GITHUB_OUTPUT
+          elif [ -n "${{ secrets.DEBUG_PRIME_USERNAME }}" ]; then
+            echo "manual_secrets=true" >> $GITHUB_OUTPUT
+          elif [ -n "${{ secrets.DEBUG_PRIME_PASSWORD }}" ]; then
+            echo "manual_secrets=true" >> $GITHUB_OUTPUT
+          else
+            echo "manual_secrets=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Read Secrets (only if not set in workflow_dispatch)
+        uses: rancher-eio/read-vault-secrets@main
+        if: ${{ steps.check_manual_secrets.outputs.manual_secrets == 'false' }}
+        with:
+          secrets: |
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials username | DOCKER_USERNAME ;
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials password | DOCKER_PASSWORD ;
+            secret/data/github/repo/${{ github.repository }}/application-collection/credentials username | APPCO_USERNAME ;
+            secret/data/github/repo/${{ github.repository }}/application-collection/credentials password | APPCO_PASSWORD ;
+            secret/data/github/repo/${{ github.repository }}/rancher-prime-registry/credentials username | PRIME_USERNAME ;
+            secret/data/github/repo/${{ github.repository }}/rancher-prime-registry/credentials password | PRIME_PASSWORD
+
+      - name: Install regsync
+        run: |
+          curl --silent --fail --location --output regsync https://github.com/regclient/regclient/releases/download/v0.5.1/regsync-linux-amd64
+          chmod +x regsync
+
+      - name: Sync Container Images
+        run: |
+          time regsync once --config --missing regsync.yaml
+        env:
+          # We use a ternary operator here. See
+          # https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#example
+          # for more details.
+          DOCKER_USERNAME: ${{ secrets.DEBUG_DOCKER_USERNAME && secrets.DEBUG_DOCKER_USERNAME || secrets.DOCKER_USERNAME }}
+          DOCKER_PASSWORD: ${{ secrets.DEBUG_DOCKER_PASSWORD && secrets.DEBUG_DOCKER_PASSWORD || secrets.DOCKER_PASSWORD }}
+          APPCO_USERNAME: ${{ secrets.DEBUG_APPCO_USERNAME && secrets.DEBUG_APPCO_USERNAME || secrets.APPCO_USERNAME }}
+          APPCO_PASSWORD: ${{ secrets.DEBUG_APPCO_PASSWORD && secrets.DEBUG_APPCO_PASSWORD || secrets.APPCO_PASSWORD }}
+          PRIME_USERNAME: ${{ secrets.DEBUG_PRIME_USERNAME && secrets.DEBUG_PRIME_USERNAME || secrets.PRIME_USERNAME }}
+          PRIME_PASSWORD: ${{ secrets.DEBUG_PRIME_PASSWORD && secrets.DEBUG_PRIME_PASSWORD || secrets.PRIME_PASSWORD }}
```

Comment:
same question - what is the purposed of the DEBUG_ secrets?

Reply:
Addressed in https://github.com/rancher/image-mirror/pull/854#discussion_r2017533560